### PR TITLE
feat(w3c):  2020 process + patent policy

### PR DESCRIPTION
**Pull request by dontcallmedom**
This pull request prepares ReSpec for transition to Process 2020, with:
* support for the two new types of CR (CR Snapshot and CR Draft) with their respective SOTD updates; this introduces a new boolean configuration option `lsMode` (for "living standard mode") which sets how the request for input is formulated in the SOTD
* support for updateable Rec (with a new boolean config option `updateableRec`) which adapts the SOTD to highlight whether a given Rec is allowed to be updated or not; it also opens the way for Recs with proposed additions and corrections (set with the array config option `revisionTypes` which can take "addition" and/or "correction" as values), in which case a `revisedRecEnd` date is expected to be configured as well
* other updates to SOTD / doc head (in particular, requiring github issues to be linked from the head of the document)

I think this includes all the changes that I know are needed. I'm assuming it is not ready yet for merging though so am looking for feedback on what would be needed to do so.

In any case, this should not be merged before September 15 (the date at which the new process is coming in operation).

### CODE REVIEW

**Comment by sidvishnoi** [comment]
@dontcallmedom Thanks for the PR!
I can't do a detailed review right now, but you can run `npm run lint -- --fix` to autofix (most of) the linting issues.

File: src/w3c/headers.js
Comment (sidvishnoi) [inline src/w3c/headers.js:321]: Can we extract this into a separate function `hasGitHubIssuesLink(conf)` or something similar?

File: src/w3c/headers.js
Comment (marcoscaceres) [inline src/w3c/headers.js:312]:
```suggestion
      "Rec-track documents must link to Github issues from their head. Please use the [`github`](https://respec.org/docs/#github) configuration option."
```

File: src/w3c/headers.js
Comment (marcoscaceres) [inline src/w3c/headers.js:794]: I think we should exclusively key off `conf.github`. Using other links this way should be considered deprecated.

File: src/w3c/templates/sotd.js
Comment (marcoscaceres) [inline src/w3c/templates/sotd.js:144]: Maybe want to check `conf.humanCREnd` here? 

```suggestion
    } else if(conf.humanCREnd) {
```

File: src/w3c/templates/sotd.js
Comment (marcoscaceres) [inline src/w3c/templates/sotd.js:185]:
```suggestion
  let reviewTarget = "";
```

File: src/w3c/templates/sotd.js
Comment (marcoscaceres) [inline src/w3c/templates/sotd.js:365]: I think `null` should be ok here... hyperHTML should then remove it.  

```suggestion
  let proposedChanges = null;
```

**Review by marcoscaceres — COMMENTED** [review]
Some initial feedback. Thanks for doing this @dontcallmedom!

File: src/w3c/headers.js
Comment (dontcallmedom) [reply src/w3c/headers.js:794]: I think deprecating using other links for github issues sounds like a good idea, but I don't think it needs to be done at the same time as the already potentially disruptive change to process 2020.

In other words, if a spec already complies with that part of pubrules using `otherLinks` rather than `github`, I don't see any reason this should be treated as an error. Having a deprecation plan for `otherLinks` referring to github issues should be handled separately (and likely through a warning rather than an error)

File: src/w3c/templates/sotd.js
Comment (dontcallmedom) [reply src/w3c/templates/sotd.js:144]: I'm not sure I understand - `humanCREnd` is needed in both conditions, and its absence is reported as an error in `w3c/headers.js`. What would this additional if brings?

**Comment by dontcallmedom** [comment]
How should we approach the potential release of this for September 15 (next Tuesday)?

**Comment by sidvishnoi** [comment]
@marcoscaceres Should we backport this to w3c-common profile, or deprecate it for real this time?

File: src/w3c/headers.js
Comment (marcoscaceres) [reply src/w3c/headers.js:794]: Yeah, I agree. Will do that as a followup.

File: src/w3c/templates/sotd.js
Comment (marcoscaceres) [reply src/w3c/templates/sotd.js:144]: Ah, I might have misunderstood.

**Comment by marcoscaceres** [comment]
> How should we approach the potential release of this for September 15 (next Tuesday)?

We can release as soon as it's merged. Seems like this should be good to go.

**Review by marcoscaceres — APPROVED** [review]
Looks good to me.

File: src/w3c/headers.js
Comment (sidvishnoi) [inline src/w3c/headers.js:608]:
```suggestion
    const unknownRevisionType = conf.revisionTypes.find(x => !revisionTypes.includes(x));
    if (unknownRevisionType) {
      pub(
        "error",
        `\`specStatus\` is "REC" with unknown revision type '${unknownRevisionType}'`
      );
    }
```

**Comment by sidvishnoi** [comment]
We might need docs for some of the new features. Adding a checklist here:

- [ ] wgPatentPolicy
- [ ] isRecTrack
- [ ] revisionTypes
- [ ] updateableRec
- [ ] revisedRecEnd
- [ ] lsMode

**Review by sidvishnoi — APPROVED** [review]

**Comment by marcoscaceres** [comment]
We should decide which of the above are private/internal and which WGs will use. I need to check.

**Pull request merged**
